crates/primitives/src/transaction/mod.rs

```diff
@@ -620,6 +620,75 @@ impl TransactionSigned {
         initial_tx.hash = initial_tx.recalculate_hash();
         initial_tx
     }
+
+    /// Decodes a eip2718 transaction where tx is [id, rlp(typedtx)]
```

Comment:
this is not quire accurate because this also decodes legacy transactions.

Docs for this need to be very on point because this entire rpc `sendRawTransaction` is not actually rlp situation is gigia confusing.
So this should make it clear that this is intended to be used in RPC.